### PR TITLE
fix: deleting card on next card properties

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

When deleting a card while the next card properties drawer is open. Deleting a card results in an error.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/27244374/todos/4951056297)

# How should this PR be QA Tested?

While a card is selected, click the Properties tab and click on the next step to open the drawer

- [x] Deleting a card while the next card properties drawer is open should be successful

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/SelectedCard/SelectedCard.tsx [hunk 1]
```diff
@@ -43,38 +43,38 @@ export function SelectedCard(): ReactElement {
...
 
   useEffect(() => {
-    setNextStep(steps?.find((step) => nextBlockId === step.id))
-  }, [steps, nextBlockId])
+    setNextStep(steps?.find((step) => currentBlock?.nextBlockId === step.id))
+  }, [steps, currentBlock])
 
   // TODO: Set as block itself for now, still need to manually set next block
   async function handleRemoveCustomNextStep(): Promise<void> {
     if (journey == null) return
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/SelectedCard/SelectedCard.tsx:58]:
```suggestion
if (currentBlock == null) return 
```

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/SelectedCard/SelectedCard.tsx [hunk 2]
```diff
@@ -43,38 +43,38 @@ export function SelectedCard(): ReactElement {
...
 
     await stepBlockDefaultNextBlockUpdate({
       variables: {
-        id,
+        id: currentBlock?.id,
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/SelectedCard/SelectedCard.tsx:61]:
```suggestion
        id: currentBlock.id,
```

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/SelectedCard/SelectedCard.tsx [hunk 3]
```diff
@@ -43,38 +43,38 @@ export function SelectedCard(): ReactElement {
...
         journeyId: journey.id,
         input: {
-          nextBlockId: id
+          nextBlockId: currentBlock?.id
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/SelectedCard/SelectedCard.tsx:64]:
```suggestion
          nextBlockId: currentBlock.id
```

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/SelectedCard/SelectedCard.tsx (hunk 1 above)
Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/SelectedCard/SelectedCard.tsx:58]: Tried this already, but still gave an error - will look into it again

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/Conditions/Conditions.tsx [hunk 4]
```diff
@@ -67,7 +67,7 @@ export function Conditions(): ReactElement {
       <ToggleOption
         heading={'Lock the next step'}
         description={'Prevent skipping of current card'}
-        checked={block.locked}
+        checked={block?.locked}
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/Conditions/Conditions.tsx:70]: What happens here if block is null? 

Can be do a block null check here, or make checked nullable and disable if null on toggleOption?

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/Cards/Cards.tsx [hunk 5]
```diff
@@ -30,26 +30,26 @@ export function Cards(): ReactElement {
   } = useEditor()
   const { journey } = useJourney()
   const theme = useTheme()
-  const { id, nextBlockId } = selectedBlock as TreeBlock<StepFields>
+  const currentBlock = selectedBlock as TreeBlock<StepFields>
 
   const nextStep: TreeBlock<StepFields> | undefined = steps?.find(
-    ({ id }) => nextBlockId === id
+    ({ id }) => currentBlock != null && currentBlock.nextBlockId === id
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/Cards/Cards.tsx:36]:
```suggestion
    ({ id }) => currentBlock?.nextBlockId === id
```

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/Conditions/Conditions.tsx [hunk 6]
```diff
@@ -67,7 +67,7 @@ export function Conditions(): ReactElement {
       <ToggleOption
         heading={'Lock the next step'}
         description={'Prevent skipping of current card'}
-        checked={block.locked}
+        checked={block != null ? block.locked : false}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Step/NextCard/Conditions/Conditions.tsx:70]:
```suggestion
        checked={block?.locked ?? false}
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Comment by tataihono** [comment]
@GeronimoJohn you still working on this?

**Comment by GeronimoJohn** [comment]
@tataihono oh yes, this is a pretty straightforward fix-ish. I don't think that I'll have time to work on the merge changes for this one during this cycle. Can re-open it at a cooldown somehow

**Pull request closed without merging**